crates/primitives/src/receipt.rs

```diff
@@ -527,10 +527,14 @@ impl ReceiptWithBloomEncoder<'_> {
             TxType::Eip7702 => {
                 out.put_u8(EIP7702_TX_TYPE_ID);
             }
-            #[cfg(feature = "optimism")]
+            #[cfg(all(feature = "optimism", not(feature = "scroll")))]
             TxType::Deposit => {
                 out.put_u8(op_alloy_consensus::DEPOSIT_TX_TYPE_ID);
             }
+            #[cfg(all(feature = "scroll", not(feature = "optimism")))]
+            TxType::L1Message => {
+                out.put_u8(reth_scroll_primitives::L1_MESSAGE_TRANSACTION_TYPE);
```

Comment:
At some point in the future we should extract this to a separate repo `scroll_alloy_consensus` such that we can use it in an alloy rpc client.